### PR TITLE
Fix rtstruct loading

### PR DESCRIPTION
**Pull request by SimonBiggs**
### PR Checklist

- [x] Brief description of changes
- [ ] Links to any relevant issues
- [ ] Required status checks are passing
- [ ] User cases if changes impact the user's experience
- [ ] `@mention` a maintainer to request a review

Fix for DICOM RT intermittently seeing a CT scan slice instead of the provided RT Structure file.

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

**Comment by SimonBiggs** [comment]
Hi @sedghi,

The checklists asks that I ping a maintainer. Would you be a good person to ping for this?

Cheers,
Simon

**Comment by sedghi** [comment]
@igoroctaviano Can you please check this out too?

File: platform/core/src/utils/dicomLoaderService.js [hunk 1]
```diff
@@ -18,14 +18,31 @@ const getImageId = imageObj => {
...
     );
-    return displaySet;
+
+    return foundDisplaySet;
+  });
+
+  const displaySet = study.displaySets.find(displaySet => {
+    return displaySet.displaySetInstanceUID === displaySetInstanceUID;
   });
+
```
Comment (igoroctaviano) [inline platform/core/src/utils/dicomLoaderService.js:31]: @SimonBiggs can you help me understand the problem and why these changes could address it? The RT sop class handler uses the first instance in its displaySet object. I'm not sure why you need to get the display set and find them again if it's always the first (previous code).

**Review by igoroctaviano — CHANGES_REQUESTED** [review]
Hi Simon, I need more context to review the pull request. Left a comment.

File: platform/core/src/utils/dicomLoaderService.js (hunk 1 above)
Comment (SimonBiggs) [reply platform/core/src/utils/dicomLoaderService.js:31]: When viewing RT structure files an error was intermittently encountered where the structure file being loaded was being returned as a CT DICOM file. This was because the first item was not the RT structure file. This code gets the UIDs of the structure file being requested and then makes sure to load the DICOM file that matches those UIDs, instead of defaulting to the first (which is intermittently incorrect).

**Review by igoroctaviano — APPROVED** [review]
LGTM. 🎖️

**Comment by sedghi** [comment]
@GitanjaliChhetri If you are happy too, can you please merge it?

**Comment by GitanjaliChhetri** [comment]
> @GitanjaliChhetri If you are happy too, can you please merge it?

yes, looks good.

**Review by sedghi — APPROVED** [review]

**Comment by sedghi** [comment]
@SimonBiggs Can you please rebase and then we can merge?

**Comment by sedghi** [comment]
base has changed, read more here https://github.com/OHIF/Viewers/issues/3477

**Comment by SimonBiggs** [comment]
> base has changed, read more here #3477

Is this going to merge, or should I close this?
